samza-core/src/main/java/org/apache/samza/task/TaskFactoryUtil.java

```diff
@@ -162,7 +164,8 @@ public static StreamApplication createStreamApplication(Config config) {
     ApplicationConfig appConfig = new ApplicationConfig(config);
     if (appConfig.getAppClass() != null && !appConfig.getAppClass().isEmpty()) {
       TaskConfig taskConfig = new TaskConfig(config);
-      if (taskConfig.getTaskClass() != null && !taskConfig.getTaskClass().isEmpty()) {
+      String taskClassName = taskConfig.getTaskClass().getOrElse(defaultValue(null));
```

Comment:
np: Given we are anyways converting it to null, how about using `taskConfig.getTaskClass().orNull`